docs/docker.md

```diff
@@ -0,0 +1,17 @@
+# Dockerfiles

```

Comment:
It looks like much of this information is a duplicate of https://github.com/exercism/automated-tests/blob/master/docs/docker.md. I'd prefer not to duplicate the information and instead link to the aforementioned document, which goal is to be a shared reference. There is some additional information in this document that I'd love to see added to that other document.